javalib/src/main/scala/java/math/BigInteger.scala

```diff
@@ -0,0 +1,1500 @@
+    val val2 = bi.abs()
+    // To avoid a possible division by zero
+    if (val1.signum() == 0) {
+      val2
+    } else if (val2.signum() == 0) {
+      val1
+    } else if (((val1.numberLength == 1) || ((val1.numberLength == 2) && (val1.digits(1) > 0))) &&
+        (val2.numberLength == 1 || (val2.numberLength == 2 && val2.digits(1) > 0))) {
+      // Optimization for small operands
+      // (op2.bitLength() < 64) and (op1.bitLength() < 64)
+      BigInteger.valueOf(Division.gcdBinary(val1.longValue(), val2.longValue()))
+    } else {
+      Division.gcdBinary(val1.copy(), val2.copy())
+    }
+  }
+
+  /**
+   * Returns the position of the lowest set bit in the two's complement
+   * representation of this {@code BigInteger}. If all bits are zero (this=0)
+   * then -1 is returned as result.
+   * <p>
+   * <b>Implementation Note:</b> Usage of this method is not recommended as the
+   * current implementation is not efficient.
+   *
+   * @return position of lowest bit if {@code this != 0}, {@code -1} otherwise
+   */
+  def getLowestSetBit(): Int = {
+    if (sign == 0) {
+      -1
+    } else {
+      // (sign != 0) implies that exists some non zero digit
+      val i = getFirstNonzeroDigit
+      (i << 5) + java.lang.Integer.numberOfTrailingZeros(digits(i))
+    }
+  }
+
+  /**
+   * Returns a hash code for this {@code BigInteger}.
+   *
+   * @return hash code for {@code this}.
+   */
+  override def hashCode(): Int = {
+    if (_hashCode != 0) {
+      _hashCode
+    } else {
+      for (i <- 0 until digits.length) {
+        _hashCode = _hashCode * 33 + (digits(i) & 0xffffffff)
+      }
+      _hashCode = _hashCode * sign
+      _hashCode
+    }
+  }
+
+  /**
+   * Returns this {@code BigInteger} as an int value. If {@code this} is too big
+   * to be represented as an int, then {@code this} % 2^32 is returned.
+   *
+   * @return this {@code BigInteger} as an int value.
+   */
+  override def intValue(): Int = sign * digits(0)
+
+  /**
+   * Tests whether this {@code BigInteger} is probably prime. If {@code true} is
+   * returned, then this is prime with a probability beyond (1-1/2^certainty).
+   * If {@code false} is returned, then this is definitely composite. If the
+   * argument {@code certainty} <= 0, then this method returns true.
+   *
+   * @param certainty tolerated primality uncertainty.
+   * @return {@code true}, if {@code this} is probably prime, {@code false}
+   *         otherwise.
+   */
+  def isProbablePrime(certainty: Int): Boolean = {
+    Primality.isProbablePrime(abs(), certainty)
+  }
+
+
+  /**
+   * Returns this {@code BigInteger} as an long value. If {@code this} is too
+   * big to be represented as an long, then {@code this} % 2^64 is returned.
+   *
+   * @return this {@code BigInteger} as a long value.
+   */
+  override def longValue(): Long = {
+    val value = if (numberLength > 1) (digits(1).toLong << 32) | (digits(0) & 0xFFFFFFFFL)
+                else digits(0) & 0xFFFFFFFFL
+    sign * value
```

Comment:
I guess you meant https://github.com/InTheNow/scala-js/blob/BigNumerics/test-suite/src/test/scala/org/scalajs/testsuite/javalib/math/BigIntegerConvertTest.scala#L375
OK